### PR TITLE
Bug 1140247 - check filepath existed or not before browse file from device storage, r=djf

### PR DESCRIPTION
**Pull request by ian-liu**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by try-server-hook** [comment]
[ian-liu Ian Liu (ian-liu)](https://github.com/ian-liu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=c1e4887faedc)

File: apps/bluetooth/js/transfer.js [hunk 1]
```diff
@@ -194,13 +196,9 @@ navigator.mozL10n.once(function showPanel() {
...
       if (blob.name) {
         // The blob has name, send the blob directly.
-        defaultAdapter.sendFile(targetDevice.address, blob);
-        var msg = 'blob is sending...';
-        debug(msg);
-        if (--numberOfTasks === 0) {
-          transferred();
-        }
-      } else {
+        msg = 'blob is sending with name...';
```
Comment (davidflanagan) [inline apps/bluetooth/js/transfer.js]: For each of these `msg = '...'` lines, I'd just write `debug('...')` instead. That way the sendFile() call does not need to have any debugging in it and you don't need to pass the argument.

File: apps/bluetooth/js/transfer.js [hunk 2]
```diff
@@ -208,27 +206,46 @@ navigator.mozL10n.once(function showPanel() {
...
+   * @memberOf transfer
+   * @param {String} address - address of target device
+   * @param {Object} blob - blob(file) to send
+   * @param {String} msg - message for debugging only
+   */
+  function sendFile(address, blob, msg) {
+    defaultAdapter.sendFile(address, blob);
+    debug(msg);
+    checkTasksComplete();
+  }
```
Comment (davidflanagan) [inline apps/bluetooth/js/transfer.js]: I like having this as a separate function, but think you should move the debug out of it.

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by try-server-hook** [comment]
[ian-liu Ian Liu (ian-liu)](https://github.com/ian-liu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=cf6ea5768860)

**Comment by ian-liu** [comment]
Revised with David's suggestion. Since all tests are passed on try-server, we can merge the patch now.

**Pull request merged**